### DISCUSSION TITLE
isce cSLC products not recognized?

### DISCUSSION THREAD

**Discussion by teinori** [isce-framework/dolphin#277] [Q&A] [answered]
I created some co-registered SLCs with `stackSentinel.py` in isce (version 2.6.3). The products seem to be `merged/SLC/*/*.slc.full`. 

I tried to load these `*.slc.full` products to dolphin (version 0.16.0) and an error occurred. The last few lines of the error message are:  
```
  File "~/micromamba/envs/dolphin2024/lib/python3.12/site-packages/dolphin/io/_core.py", line 248, in get_raster_xysize
    ds = gdal.Open(fspath(filename))
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/micromamba/envs/dolphin2024/lib/python3.12/site-packages/osgeo/gdal.py", line 5985, in Open
    return _gdal.Open(*args)
           ^^^^^^^^^^^^^^^^^
  RuntimeError: `~/fun/insar/okmok/input_slcs/20220616.slc.full' not recognized as a supported file format.
```
Could this just be a `gdal` problem? Are extra steps needed before I can put the isce stack output into dolphin for further processing? Thank you.  

Edit: I tried the burst c-slc products under isce directory `coreg_secondarys/*/IW*/burst*.slc`. They are not recognized as a supported file format in dolphin either.

**Comment by scottstanie** [comment, marked as answer]
Hi, this does look like a problem between the interaction with GDAL and ISCE2.

Can you run `gdalinfo <name of SLC file>` on the SLCs and get a useful output?

If not, did you try pointing it at the `.vrt` files (assuming they are present for these SLCs)?

**Reply by teinori** [reply]
Thank you for your response, Doctor. Your perception is right. `gdalinfo <name of SLC file>` gave the same error (not recognized as a supported file format), but when given `.vrt` files as cSLC input, dolphin is able to read them correctly and run all following steps. This question is marked solved.

**Reply by pyyff** [reply]
Hello, could you please explain in detail how to proceed? I have encountered a similar issue and would be very grateful.
